fop/fom.c

```diff
@@ -508,6 +508,7 @@ static void queueit(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 
 	addb2_introduce(fom);
 	m0_fom_locality_inc(fom);
+	m0_atomic64_dec(&fom->fo_service->rs_fom_queued);
```

Comment:
Shouldn't it be decreased after the fom is executed already? Otherwise, we will execute the fom when its service is shut down already - is this even legal?

Reply:
Reqh service stop also checks for locality counter for the corresponding service which was increment just above this function.

Reply:
Moved it after fom service state is checked.